### DISCUSSION TITLE
Are you using Rx.NET?

### DISCUSSION THREAD

**Discussion by HowardvanRooijen** [dotnet/reactive#2031] [Announcements]
Hi Community!

@idg10 is presenting [Modernizing Rx.NET](https://www.dotnetconf.net/agenda#536837) at .NET Conf on Thursday, 16 November 13:30 - 14:00 GMT.

We'd like to feature orgs that are using Rx.NET

Please post your org name, a link to a logo, and if you could mention your use-case, that would be awesome too!

Thanks!

**Comment by apobekiaris** [comment]
I specialize in creating open-source modules for DevExpress ExpressApp (XAF), showcased at [eXpandFramework](https://github.com/eXpandFramework/Reactive.XAF). While [DevExpress XAF](https://devexpress.com/xaf) is inherently an MVC multi-platform framework, my repository takes a distinct path, adopting an entirely event-driven approach powered by RX.NET—eschewing traditional MVC controllers. The cornerstone of this initiative is the [Reactive.XAF module](https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Reactive), which lays the groundwork with its API for developing the [suite of framework modules](https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules). Embracing RX.NET has reshaped my coding practices for all my applications, and I've steadfastly adhered to this paradigm for years without looking back.

![](https://user-images.githubusercontent.com/159464/66713086-c8c5a800-edae-11e9-9bc1-73ffc0c215fb.png)

To update your response in the GitHub discussion at [dotnet/reactive #2031](https://github.com/dotnet/reactive/discussions/2031), you can add the following information:

---

**Update**: In the upcoming December release of DevExpress, the OutlookInspired demo will feature the use of RX.NET. This implementation is specifically for creating an integration test API used to test the demo. This demo, which is a part of the 23.2.2 release, is currently available for preview at [DevExpress Examples](https://github.com/DevExpress-Examples/How-to-create-a-multitenancy-application/tree/23.2.2+). This addition further demonstrates the versatility and practical application of RX.NET in real-world scenarios, especially in the context of testing and validating complex application behaviors.

---

**Comment by laurentkempe** [comment]
At [Innoveo](https://innoveo.com/) we are using Rx.NET in the Business Canvas. Non-technical people "Use Business Canvas to quickly and easily configure apps, products, and workflows in a straightforward, visual environment". Business Canvas is a WPF application using the MVVM pattern and leverage Rx.NET on different layers from the infrastructure to load a project, to the event driven validation of configurations and in some part of the view models.

![innoveo-logo-primary-BLACK](https://github.com/dotnet/reactive/assets/272612/99fd1127-93b0-491b-9e88-285bf5b764d9)

**Comment by glopesdev** [comment]
We have been developing the [Bonsai](https://bonsai-rx.org/) visual reactive programming language for more than 10 years built entirely on Rx.NET. Our goal is to drastically reduce the barrier to entry for scientific experimentation, where data acquisition and control of asynchronous streams is becoming the norm.

The backbone of Bonsai is a general-purpose functional programming language with type inference where the entire application is a single reactive query! The compiler emits an expression tree specifying an observable sequence, so programming in Bonsai is programming directly with Rx. There are now thousands of people becoming exposed and proficient in Rx without even realizing it simply by practicing visual reactive programming in Bonsai and skimming through our [docs](https://bonsai-rx.org/docs/articles/observables.html), which include marble diagrams and hands-on tutorials.

The language is increasingly used for real-time scientific experimentation, especially in biology and neuroscience, with [over 50 packages](https://www.nuget.org/profiles/bonsai) integrating everything from industrial cameras, DAQs, motorized stages, Arduino, networking, real-time computer graphics, machine-learning inference on streams, EEG, and other specialized electrophysiology systems for neuroscience. Bonsai integrates directly with the .NET ecosystem and is fully extensible using NuGet packages, which supports an increasingly rich set of community extensions.

<img alt="bonsai-logo" src="https://github.com/dotnet/reactive/assets/5315880/3e24abbd-89ad-4200-99f2-4f643a0540a0" />
